public/app/features/explore/Graph.tsx

```diff
@@ -143,52 +174,64 @@ export class Graph extends PureComponent<GraphProps, GraphState> {
     );
   };
 
-  draw() {
-    const { range, size, userOptions = {} } = this.props;
+  onToggleSeries = (series: TimeSeries, exclusive: boolean) => {
+    this.setState(
+      (state, props) => {
+        const { data } = props;
+        const { hiddenSeries } = state;
+        const hidden = hiddenSeries.has(series);
```

Comment:
not sure how well this set will work, will it not be using reference equality? So next refresh, with new TimeSeries instances it won't match anymore?